### ISSUE TITLE
Unable to extract files from ZIP file with version 45

### ISSUE THREAD

**Issue by bjorn** [nezticle/qtcompress#1]
This code apparently only supports ZIP files up to version "20", and the file I tried to read with it appears to be version "45". This results in the message:
```
QZip: .ZIP specification version 45 implementationis needed to extract the data.
```
I know this library is unmaintained and only meant at a starting point for a full featured ZIP file reader/writer for Qt, but I thought I'd document this shortcoming here for reference.

**Comment by nezticle** [comment]
Good catch.  It should be possible to update the version of zlib in src/3rdparty, but I've not tried to do so and there may be some problems I've not forseen

**Comment by bjorn** [comment]
Actually I'm using the zlib that ships with Qt which appears to be version 1.2.5 as well. I'm not sure why Qt ships with such an old zlib header, but on the other hand I doubt a little that this would be the reason for the incompatibility since there's just patch releases in between.

**Comment by bjorn** [comment]
According to:

https://pkware.cachefly.net/webdocs/APPNOTE/APPNOTE-6.3.3.TXT

The version 45 means "File uses ZIP64 format extensions".

**Comment by nezticle** [comment]
ah, so likely some "real" work to change how we are using zlib

**Comment by bjorn** [comment]
Well, I've hacked it up to get it to work with my zip file. Will look into opening a pull request later, though it'll be mainly to show the rough changes because for sure it won't have all the proper error handling in place.

In my case I did not need to change anything regarding the way zlib is used, because the actual ZIP file did not contain any files larger than 4 GB. Rather it was for whatever reason using this ZIp64 extension to store the compressed and uncompressed file systems in an extra field. This is valid according to the standard, and it seems to be not too complicated to support that.

I think actually supporting larger amounts of data would be really hard, maybe impossible with the current API. The reason being that QByteArray can't store more than about 2 GB of data, or am I missing something? It just uses 'int' for its size, at least.

Also, zip version 2.7 added a "File is a patch data set" flag that will now probably need to be reported as unsupported.

**Comment by bjorn** [comment]
For future reference, this could be useful for implementing full ZIP64 support since it contains some useful links and patches to OpenJDK that added support for that:

https://blogs.oracle.com/xuemingshen/entry/zip64_support_for_4g_zipfile
